src/preview.ts

```diff
@@ -77,7 +77,7 @@ export class PreviewServer {
     Telemetry.record({event: "preview", step: "start"});
     const server = createServer();
     if (port === undefined) {
-      for (port = 3000; true; ++port) {
+      for (port = 3000; port < 49152; ++port) {
```

Comment:
Can you make a `MAX_PORT` constant for this since it’s in a couple places? And where does it come from — seems like 46,152 attempts might be a bit too many lol?

Reply:
I just wanted to avoid the infinite loop and needed a reasonable target. 49152 is the start of the [Dynamic, private or ephemeral ports](https://en.wikipedia.org/wiki/Ephemeral_port) range. (See also https://en.wikipedia.org/wiki/Registered_port).

It's a lot of tests, but it's almost instant even if we need to cover the whole range.

Reply:
I added MAX_PORT

Reply:
I'm late here, but how would you feel about letting the OS choose a port when there is a conflict? If you request the port 0 you get a "random" free port.

Reply:
seems interesting (& I didn't know that); we could count up to 3010 or 3100, and then do that trick… not sure how you get back the actual port number though?

Reply:
After you start listening the `server.address()` method has information about that, though it's a little more complicated than just the port. Here's what we do in project-builder:

```ts
httpServer.listen(config.port, "127.0.0.1", () => {
  const address = httpServer.address();
  if (!address || typeof address === "string")
    throw new Error(`Unexpected server address: ${address}`);
  console.log(`Listening on http://${address.address}:${address.port}/`);
});
```